pdr_backend/lake/duckdb_data_store.py

```diff
@@ -243,34 +288,79 @@ def query_data(self, query: str) -> Optional[pl.DataFrame]:
+        Builds the query string to perform the operation and returns it
         @arguments:
-            temp_table - The temporary table object
-            permanent_table_name - The name of the permanent table.
+            drop_table_name - The table to drop records from.
+            ref_table_name - The table to reference for the IDs to drop.
+        @example:
+            get_query_drop_records_from_table_by_id("bronze_pdr_slots", "update_pdr_slots")
+        """
+
+        # Return the query string
+        return f"""
+        DELETE FROM {drop_table_name}
+        WHERE ID IN (
+            SELECT DISTINCT ID 
+            FROM {ref_table_name}
+        );
+        """
+
+    @enforce_types
+    def drop_records_from_table_by_id(self, drop_table_name: str, ref_table_name: str):
+        """
+        Drop the records from the table by the provided IDs.
+        @arguments:
+            table_name - A unique name for the table.
+            ids - The list of IDs to drop.
+        @example:
+            drop_records_from_table_by_id("bronze_pdr_slots", "update_pdr_slots")
+        """
+        query = self.get_query_drop_records_from_table_by_id(
+            drop_table_name, ref_table_name
+        )
+        self.execute_sql(query)
+
+    @enforce_types
+    def get_query_move_table_data(self, from_table, to_table):
+        """
+        Builds the query string to perform the operation and returns it
+        @arguments:
+            from_table - where we'll be taking data from
+            to_table - wehere we'll be inserting data
         @example:
             move_table_data("temp_people", "people")
         """
 
         # Check if the table exists
         table_names = self.get_table_names()
+        assert (
+            from_table.table_name in table_names
+        ), f"Table {from_table.table_name} is "
 
-        if temp_table.fullname not in table_names:
-            logger.info(
-                "move_table_data - Table %s does not exist", temp_table.fullname
-            )
-
-        permanent_table_name = permanent_table.fullname
         # check if the permanent table exists
-        if permanent_table_name not in table_names:
-            # create table if it does not exist
-            self.execute_sql(
-                f"CREATE TABLE {permanent_table_name} AS SELECT * FROM {temp_table.fullname}"
-            )
-            return
+        if to_table.table_name not in table_names:
+            return f"CREATE TABLE {to_table.table_name} AS SELECT * FROM {from_table.table_name};"
 
         # Move the data from the temporary table to the permanent table
-        self.execute_sql(
-            f"INSERT INTO {permanent_table_name} SELECT * FROM {temp_table.fullname}"
+        return (
+            f"INSERT INTO {to_table.table_name} SELECT * FROM {from_table.table_name};"
         )
+
+    @enforce_types
+    def row_count(self, table_name: str) -> int:
```

Comment:
I think row_count is also one of the things we removed in main. See above for update_data.

Reply:
I have specifically added these things for tooling and validations around the CLI... these operations are much faster done through the DB. 

Please operate the lake, become a heavy user, debug incremental pipelines, and then make decisions as to what functionality to drop.